### PR TITLE
Qute - force getters access for panache entities

### LINKED ISSUE #14918: Qute does not invoke getters generated by Hibernate Panache
**Describe the bug**
Because Qute accesses properties using the public field access, lazy loading is never triggered when Qute templates are rendered.

**Expected behavior**
When iterating a lazy (default) `@OneToMany` relationship, the entities should be queried from the DB (implemented in Panache when the getter method is invoked).

**Actual behavior**
Lazy collections in `@OneToMany` associations are never triggered

**To Reproduce**
Run the following project: 

[code-with-quarkus.zip](https://github.com/quarkusio/quarkus/files/5947031/code-with-quarkus.zip)

And hit http://localhost:8080/fruits. You should see `Apple`.

Go to the `Fruit` entity an uncomment the getter + private field and hit the URL again. You'll see `Apple` and two worms

### PR DESCRIPTION
**Pull request by mkouba**

### CODE REVIEW

File: extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/QuteProcessor.java [hunk 1]
```diff
@@ -821,7 +821,20 @@ public boolean test(String name) {
...
+        entitySuperClasses.add(DotName.createSimple("io.quarkus.hibernate.orm.panache.PanacheEntityBase"));
+        entitySuperClasses.add(DotName.createSimple("io.quarkus.mongodb.panache.PanacheMongoEntity"));
+        entitySuperClasses.add(DotName.createSimple("io.quarkus.mongodb.panache.PanacheMongoEntityBase"));
+        ValueResolverGenerator.Builder builder = ValueResolverGenerator.builder()
+                .setIndex(index)
+                .setForceGettersPredicate(new Predicate<ClassInfo>() {
+                    @Override
+                    public boolean test(ClassInfo clazz) {
+                        DotName superName = clazz.superName();
+                        return superName != null && entitySuperClasses.contains(superName);
```
Comment (FroMage) [inline extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/QuteProcessor.java:835]: This is forgetting HR with Panache, mongo reactive, and kotlin versions. Also this only tests for direct superclasses, but there are entire hierarchies of such entities. I fear this is going to be too fragile. Perhaps we should either use a panache build item or make a new build item in core for this sort of case?

**Review by FroMage — CHANGES_REQUESTED** [review]

File: extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/QuteProcessor.java (hunk 1 above)
Comment (mkouba) [reply extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/QuteProcessor.java:835]: We could use the `PanacheEntityClassesBuildItem` if we create an SPI module. I'm not so sure about moving this build item to the core...

Comment (FroMage) [reply extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/QuteProcessor.java:835]: SPI module is fine, but perhaps the `-common` module is already an SPI?

Comment (mkouba) [reply extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/QuteProcessor.java:835]: Hm, I can try...

Comment (mkouba) [reply extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/QuteProcessor.java]: It seems to work :-)

File: extensions/qute/deployment/pom.xml [hunk 2]
```diff
@@ -25,6 +25,11 @@
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-arc-deployment</artifactId>
         </dependency>
+        <!-- Panache SPI-like module to collect entity classes -->
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-panache-common-deployment</artifactId>
```
Comment (gastaldi) [inline extensions/qute/deployment/pom.xml:31]: Can this extension be marked optional and be included only when Panache is used with Qute somehow?

Comment (FroMage) [reply extensions/qute/deployment/pom.xml:31]: It's an SPI module, it doesn't depend on anything.

**Review by FroMage — APPROVED** [review]
LGTM

File: extensions/qute/deployment/pom.xml (hunk 2 above)
Comment (mkouba) [reply extensions/qute/deployment/pom.xml:31]: This extension should not contain any logic. TBH I'm not sure what happens if I add `<optional>true</optional>`...

Comment (mkouba) [reply extensions/qute/deployment/pom.xml:31]: @aloubyansky Does `<optional>true</optional>` in a deployment module POM affect the quarkus build in any way?

Comment (aloubyansky) [reply extensions/qute/deployment/pom.xml:31]: @mkouba optional dependencies of an extension will not be added to the application build classpath

Comment (mkouba) [reply extensions/qute/deployment/pom.xml:31]: In this case, the dependency is an SPI-like `-deployment` module and we use a class from this module in a build step, i.e. the class needs to be added on the app build class path. So I think it's ok not to add `<optional>true</optional>` here, right?

Comment (aloubyansky) [reply extensions/qute/deployment/pom.xml:31]: That sounds like it doesn't qualify as optional at all.

**Pull request merged**